engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationFieldsBuilder.kt

```diff
@@ -66,9 +65,8 @@ internal object NadelHydrationFieldsBuilder {
                 if (objectTypesAreNotReturnedByActorField) {
                     null
                 } else {
-                    childField.toBuilder()
-                        .objectTypeNames(childField.objectTypeNames.filter { it in actorFieldOverallObjectTypeNames })
-                        .build()
+                    childField.objectTypeNames.removeIf { it !in actorFieldOverallObjectTypeNames }
+                    childField
```

Comment:
Can we keep the builder and do `clearObjectTypesNames`. Just because we don't mutate it anywhere else in the codebase, so I'd prefer to leave it as effectively immutable. I know we're mutating a freshly cloned object, but I don't want others to see this and think it's ok to mutate it in other places.